lib/matplotlib/tri/_tripcolor.py

```diff
@@ -80,10 +80,7 @@ def tripcolor(ax, *args, alpha=1.0, norm=None, cmap=None, vmin=None,
                 "tripcolor() missing 1 required positional argument: 'c'; or "
                 "1 required keyword-only argument: 'facecolors'")
         elif len(args) > 1:
-            _api.warn_deprecated(
-                "3.6", message=f"Additional positional parameters "
-                f"{args[1:]!r} are ignored; support for them is deprecated "
-                f"since %(since)s and will be removed %(removal)s")
+            raise TypeError(f"Unexpected positional parameters: {args[1:]!r}")
```

Comment:
Should use `_api.nargs_error`.

Reply:
This is a confusing case for the "takes X, given Y" number of parameters. The expected number of parameters varies based on which call signature you are using, and those get normalized out by `Trangulation.get_from_args_and_kwargs` so `args` the variable needs to have only one value, but what you pass could be 

- 1: if you pass triangulation and facecolors(as kwarg)
- 2: either triangulation and c or x,y and facecolors(as kwarg)
- 3: x,y,c

In the cases where facecolors is passed, adding `c` will warn that it is ignored, but not error (and in fact, I'm realizing now will not error even if _more_ parameters are passed...) So the facecolors cases won't get to this codepath, which narrows it down to the triangulaion/c or x/y/c case, but we still don't know which version was used (without adding a boolean flag or something)

There is even a technical way to pass 4 parameters validly discussed (but discouraged) in the docstring.

Reply:
Well, `nargs_error` doesn't need to take a fixed number, but could take, e.g., `1-4` as the number of expected (which we do elsewhere already). However, it seems like you are saying that we're more specific here about extra arguments depending on other inputs, and don't want to limit to just the expected/actual message.